### PR TITLE
Configure ProcDumpCollector to attach dumps to test results

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2844

Regression? no

## Description

Set VSEng's procdumpcollector settings as per their recommendation

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by donnie-msft — APPROVED** [review]
I looked for this PR in the test pipeline and didn't see any failures. I don't see why it wouldn't, but curious if you tested whether failed test runs do include dumps now?

File: build/template.runsettingsproj [hunk 1]
```diff
@@ -140,7 +140,8 @@
     <DataCollector Condition="'$(ProcDumpCollector)' == 'enabled'" Include="datacollector://Microsoft/DevDiv/TestExtensions/ProcDumpCollector/v1">
       <FriendlyName>ProcDump Collector</FriendlyName>
       <Configuration>
-        <RootDumpDirectory>C:\Test\Logs\Dumps</RootDumpDirectory>
```
Comment (jeffkl) [inline build/template.runsettingsproj:143]: 🤣

**Review by jeffkl — APPROVED** [review]
💯

**Comment by zivkan** [comment]
> I looked for this PR in the test pipeline and didn't see any failures. I don't see why it wouldn't, but curious if you tested whether failed test runs do include dumps now?

Yes, it was tested. Look at the OptProf pipeline, the runs where the branch name matches this PR's branch name.

**Review by jebriede — APPROVED** [review]

**Pull request merged**